src/DynamoSandbox/Program.cs

```diff
@@ -229,21 +230,43 @@ private static void PreloadShapeManager(ref string geometryFactoryPath, ref stri
             preloaderLocation = preloader.PreloaderLocation;
         }
 
+        [DllImport("msvcrt.dll")]
+        public static extern int _putenv(string env);
+
         [STAThread]
         public static void Main(string[] args)
         {
             DynamoViewModel viewModel = null;
+            var supportedLocale =
+            new HashSet<string>(new[]
+                {
+                    "cs-CZ", "de-DE", "en-US", "es-ES", "fr-FR", "it-IT",
+                    "ja-JP", "ko-KR", "pl-PL", "pt-BR", "ru-RU", "zh-CN", "zh-TW"
+                });
 
             try
             {
                 var cmdLineArgs = CommandLineArguments.FromArguments(args);
+                string locale;
 
                 if (!string.IsNullOrEmpty(cmdLineArgs.Locale))
                 {
                     // Change the application locale, if a locale information is supplied.
                     Thread.CurrentThread.CurrentUICulture = new CultureInfo(cmdLineArgs.Locale);
                     Thread.CurrentThread.CurrentCulture = new CultureInfo(cmdLineArgs.Locale);
+                    locale = cmdLineArgs.Locale;
+                }
+                else
+                {
+                    // in case no language is specified, the locale of libG should be that of the operating
+                    // system. If that locale is not supported by Dynamo, default to en-US.
+                    var systemLocale = CultureInfo.InstalledUICulture.ToString();
+                    locale = supportedLocale.Contains(systemLocale) ? systemLocale : "en-US";
```

Comment:
Is this case sensitive comparison? I definitely do not want a lower-case locale to fail us, perhaps:

```csharp
supportedLocale.Any(x => x.Compare(systemLocal, BlaBla.CaseInsensitive))
```

And do test it out for real, my Linq is rusty.